x-pack/plugins/security_solution/server/usage/detections_helpers.ts

```diff
@@ -0,0 +1,148 @@
+import { DetectionRulesUsage, MlJobsUsage } from './detections';
+import { isJobStarted } from '../../common/machine_learning/helpers';
+
+interface DetectionsMetric {
+  isElastic: boolean;
+  isEnabled: boolean;
+}
+
+const isElasticRule = (tags: string[]) => tags.includes(`${INTERNAL_IMMUTABLE_KEY}:true`);
+
+const initialRuleUsage: DetectionRulesUsage = {
+  detection_rules_custom_enabled: 0,
+  detection_rules_custom_disabled: 0,
+  detection_rules_elastic_enabled: 0,
+  detection_rules_elastic_disabled: 0,
+};
+
+const initialMlJobUsage: MlJobsUsage = {
+  ml_jobs_custom_enabled: 0,
+  ml_jobs_custom_disabled: 0,
+  ml_jobs_elastic_enabled: 0,
+  ml_jobs_elastic_disabled: 0,
+};
+
+export const buildRuleUsage = (rulesMetrics: DetectionsMetric[]): DetectionRulesUsage =>
+  rulesMetrics.reduce((stats, { isEnabled, isElastic }) => {
+    if (isEnabled && isElastic) {
+      return {
+        ...stats,
+        detection_rules_elastic_enabled: stats.detection_rules_elastic_enabled + 1,
+      };
+    } else if (!isEnabled && isElastic) {
+      return {
+        ...stats,
+        detection_rules_elastic_disabled: stats.detection_rules_elastic_disabled + 1,
+      };
+    } else if (isEnabled && !isElastic) {
+      return {
+        ...stats,
+        detection_rules_custom_enabled: stats.detection_rules_custom_enabled + 1,
+      };
+    } else if (!isEnabled && !isElastic) {
+      return {
+        ...stats,
+        detection_rules_custom_disabled: stats.detection_rules_custom_disabled + 1,
+      };
+    } else {
+      return stats;
+    }
+  }, initialRuleUsage);
+
+export const buildMlJobUsage = (jobMetrics: DetectionsMetric[]): MlJobsUsage =>
+  jobMetrics.reduce((stats, { isEnabled, isElastic }) => {
+    if (isEnabled && isElastic) {
+      return {
+        ...stats,
+        ml_jobs_elastic_enabled: stats.ml_jobs_elastic_enabled + 1,
+      };
+    } else if (!isEnabled && isElastic) {
+      return {
+        ...stats,
+        ml_jobs_elastic_disabled: stats.ml_jobs_elastic_disabled + 1,
+      };
+    } else if (isEnabled && !isElastic) {
+      return {
+        ...stats,
+        ml_jobs_custom_enabled: stats.ml_jobs_custom_enabled + 1,
+      };
+    } else if (!isEnabled && !isElastic) {
+      return {
+        ...stats,
+        ml_jobs_custom_disabled: stats.ml_jobs_custom_disabled + 1,
+      };
+    } else {
+      return stats;
+    }
+  }, initialMlJobUsage);
+
+export const fetchRules = async (
+  index: string,
+  callCluster: LegacyAPICaller
+): Promise<DetectionsMetric[]> => {
+  let ruleMetrics: DetectionsMetric[] = [];
+  const ruleSearchOptions: SearchParams = {
+    body: { query: { bool: { filter: { term: { 'alert.alertTypeId': SIGNALS_ID } } } } },
+    filterPath: ['hits.hits._source.alert.enabled', 'hits.hits._source.alert.tags'],
+    ignoreUnavailable: true,
+    index,
+    size: 10000, // elasticsearch index.max_result_window default value
```

Comment:
Are we ever likely to exceed this limit?

Reply:
No, it doesn't seem likely that a user'd have 10k rules, but this ensures that if they did, we'd tally the first 10k instead of receiving an error and returning 0s.

Reply:
This was used in another collector, but if you think there's risk here I'm happy to remove it!

Reply:
I think it's fine. And I hope they don't have more than 10k rules 😅. I was really just curious if there was a way beyond it beyond `scroll`. I ran into na issue previously where I needed to set this max just to get all of the data I wanted (I think ES has a default max it returns?), but it looks fine.